sphinx_gallery/gen_rst.py

```diff
@@ -741,7 +757,8 @@ def _get_last_repr(capture_repr, ___):
             last_repr = None
             repr_meth = None
         else:
-            if isinstance(last_repr, str):
+            if (isinstance(last_repr, str) or
+                    isinstance(last_repr, bytes) and meth in ('_repr_png',)):
```

Comment:
Question: could you explain why we need the `isinstance(last_repr, bytes) and meth in ('_repr_png',)` ? Would we not want to stop if `meth` was `'_repr_svg_'` ?

Reply:
svg will return `str` so it will stop.

Reply:
Thanks. My question is more why do we need `isinstance(last_repr, bytes) and meth in ('_repr_png',)` instead of just `(isinstance(last_repr, (str, bytes))`

Reply:
What other supported methods could give us `bytes` or is this in case we add new supported methods in future?

Reply:
I had `(isinstance(last_repr, (str, bytes))` first, but then I thought that maybe there was a reason to check for `str` that I could not figure out. (Although unlikely that `_repr_html_` will return bytes?)

Yes, the tuple is for future extensions. Now, I added JPEG as well.

Reply:
I had a look at my old PR and I don't think there was any special reason to check for `str` except for maybe safety 'just in case' something weird came out of one of the methods, but as you say it is unlikely `_repr_html_` will return `bytes`, so I think it would be safe to use `(isinstance(last_repr, (str, bytes))` - and even this would be just for safety I think